orchagent/routeorch.cpp

```diff
@@ -1317,47 +1317,87 @@ bool RouteOrch::removeNextHopGroup(const NextHopGroupKey &nexthops)
     return true;
 }
 
+void RouteOrch::addNextHopRoute(const NextHopKey& nextHop, const RouteKey& routeKey)
+{
+    auto it = m_nextHops.find((nextHop));
+
+    if (it != m_nextHops.end())
+    {
+        if (it->second.find(routeKey) != it->second.end())
+        {
+            SWSS_LOG_INFO("Route already present in nh table %s",
+                          routeKey.prefix.to_string().c_str());
+            return;
+        }
+
+        it->second.insert(routeKey);
+    }
+    else
+    {
+        set<RouteKey> routes;
+        routes.insert(routeKey);
+        m_nextHops.insert(make_pair(nextHop, routes));
+    }
+}
+
+void RouteOrch::removeNextHopRoute(const NextHopKey& nextHop, const RouteKey& routeKey)
+{
+    auto it = m_nextHops.find((nextHop));
+
+    if (it != m_nextHops.end())
```

Comment:
Do we need to add some log if the nextHop is not found?

Reply:
added